drivers/blktap2.py

```diff
@@ -1467,6 +1467,7 @@ def attach(self, sr_uuid, vdi_uuid, writable, activate = False):
         # Return backend/ link
         back_path = self.BackendLink.from_uuid(sr_uuid, vdi_uuid).path()
         struct = { 'params': back_path,
+                   'o_direct': util.read_caching_is_restricted(self._session),
```

Comment:
I think this is a race condition. The blktap driver gets hold of this value here and it's this that's passed to the Tapdisk class: https://github.com/siddharthv/sm/blob/CP-10986/drivers/blktap2.py#L1564-L1565.

In between these two places the licensing could have changed meaning the tapdisk is forked with one option but the return data says the opposite.